### PR TITLE
Liquid Filter list_items does not respect order if Ordering is enabled

### LINKED ISSUE #10682: Liquid Filter list_items does not respect order if Ordering is enabled
### Describe the bug
Liquid Filter list_items does not respect order if Ordering is enabled
### To Reproduce
1. Create a content type with list part
2. At list part, enable ordering
3. Change the order of the items

### Expected behavior
`{{ list_items }}` liquid filter should return list items in predefined order

### PR DESCRIPTION
**Pull request by urbanit**
Fixes #10682

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Lists/Helpers/ListQueryHelpers.cs [hunk 1]
```diff
@@ -19,10 +19,11 @@ internal static Task<int> QueryListItemsCountAsync(ISession session, string list
                     .CountAsync();
         }
 
-        internal static Task<IEnumerable<ContentItem>> QueryListItemsAsync(ISession session, string listContentItemId, Expression<Func<ContentItemIndex, bool>> itemPredicate = null)
+        internal static Task<IEnumerable<ContentItem>> QueryListItemsAsync(ISession session, string listContentItemId, Expression<Func<ContentItemIndex, bool>> itemPredicate = null, Expression<Func<ContainedPartIndex, object>> partPredicate = null)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Lists/Helpers/ListQueryHelpers.cs:22]: If there is nothing else to order by on ContainedPart I assume a `bool order` would be sufficient?

Comment (urbanit) [reply src/OrchardCore.Modules/OrchardCore.Lists/Helpers/ListQueryHelpers.cs:22]: I was thinking of adding a part for visits and analytics... So, liquid tag could be used with a custom filter...

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Lists/Helpers/ListQueryHelpers.cs:22]: It's an internal method so not sure how you would do that?

Plus here you have also added a default ordering, which is necesary?

I think the razor helper `ListOrchardHelperExtensions` needs the same update

File: src/OrchardCore.Modules/OrchardCore.Lists/Liquid/ListItemsFilter.cs [hunk 2]
```diff
@@ -29,10 +37,36 @@ public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArgument
             {
                 listContentItemId = input.ToStringValue();
             }
+            bool useOrder = false;
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Lists/Liquid/ListItemsFilter.cs:40]: Would be so much simpler with just the `bool order` property, can be try by default. I assume that if Order is disabled on the ListPart then it's not an issue to order on the field? So we don't have to check for the default setting. Or just use what the filter argument states if defined (or true if not defined)

**Comment by sebastienros** [comment]
I'd like the `bool order` option instead of a predicate.

**Comment by urbanit** [comment]
Will do...

**Comment by Piedone** [comment]
Is this something you'd like to revisit any time soon @urbanit or should we close?

**Comment by Piedone** [comment]
Closing due to inactivity.

**Pull request closed without merging**